compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/test/conv2d_to_winograd.mlir

```diff
@@ -85,3 +85,47 @@ func.func @conv_16433136_nchw_fchw(%arg0: tensor<1x4x16x16xf32>, %arg1: tensor<1
 // CHECK-SAME:     tensor<1x16x18x18xf32> to tensor<1x16x14x14xf32>
 // CHECK:        return %[[EXTRACTED_SLICE]] : tensor<1x16x14x14xf32>
 // CHECK:      }
+
+// -----
+
+func.func @conv_promoted(%arg0: tensor<1x16x16x4xf16>, %arg1: tensor<3x3x4x16xf16>, %arg2: tensor<1x14x14x16xf32>) -> tensor<1x14x14x16xf32> {
```

Comment:
[optional] Use `mixed_types` instead of `promoted`. IMO, `promote` could have more than one scenario. E.g., there is "promote buffer" in MLIR codebase. `mixed_type` is more straightforward and clean to me.